### DISCUSSION TITLE
allow specifying the path to the dumpbin executable when using VCPKG_APPLOCAL_DEPS

### DISCUSSION THREAD

**Discussion by georg-emg** [microsoft/vcpkg#17928] [Ideas]
****

**Is your feature request related to a problem? Please describe.**
When creating a cmake toolchain under Windows, all the relevant executables can be specified using cmake variables like `CMAKE_CXX_COMPILER` `CMAKE_RC_COMPILER` `CMAKE_MT` etc. This allows for writing a toolchain file that can be run from a regular shell without calling `vcvarsall.bat`, which makes writing CI pipelines a lot easier. These toolchain files can include the `vcpkg.cmake` toolchain file to take advantage of the advanced vcpkg features. When trying to use the `VCPKG_APPLOCAL_DEPS` feature, however, the script `applocal.ps1` will attempt to call `dumpbin` without a path. This results in an error if `dumbin` is not in the PATH, i.e. if `vcvarsall.bat` has not been called.

**Proposed solution**
The `vcpkg.cmake` toockain file could provide a cache variable `VCPKG_DUMPBIN` by calling `find_program(VCPKG_DUMPBIN, dumpbin)`, and pass the result on to `applocal.ps1` as a parameter. This would allow for specifying the path to `dumpbin` from the cmake command line, or from a toolchain file.

**Describe alternatives you've considered**
The two possible workarounds are:
- require `vcvarsall.bat` to be called before building a project that uses vcpkg. This kind of defeats the purpose of a cmake toolchain file, though, which should describe the whole build environment.
- set `VCPKG_APPLOCAL_DEPS` to `OFF`, which would be a pity.

**Additional context**
This is just a small change to `scripts/buildsystems/vcpkg.cmake` and `scripts/buildsystems/msbuild/applocal.ps1` each. I have a working patch and could submit a PR if there is interest.

**Comment by Milerius** [comment]
+1 ^

**Comment by georg-emg** [comment]
Hi, I saw that there was changes in scripts/buildsystems/msbuild/applocal.ps1 to the logic that looks for dumpbin. With the new changes, I think it would be preferrable if applocal.ps1 finds dumpbin.exe in a Visual Studio installation using the same logic that is used by bootstrap-vcpkg.bat.

**Comment by bowie7070** [comment]
@georg-emg Do you still have the PR you mentioned? If so, can you provide a link to it. Thanks

**Comment by pdehne** [comment]
I used your suggestions and have added the ability to configure a custom dumpbin path to https://github.com/pdehne/vcpkg. This 
works for me with Visual Studio 2026.

It specifically fixes builds on Windows with Ninja when using the CMake Toolchain files for Windows (https://github.com/MarkSchofield/WindowsToolchain).

I used the following configuration.

CMakePresets.json:

```json
{
  "version": 2,
  "configurePresets": [
    {
      "name": "vcpkg",
      "generator": "Ninja",
      "binaryDir": "${sourceDir}/build",
      "cacheVariables": {
        "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
      }
    }
  ]
}
```

CMakeUserPresets.json:

```json
{
  "version": 2,
  "configurePresets": [
    {
      "name": "vcpkg-msvc",
	  "inherits": "vcpkg",
      "cacheVariables": {
        "VCPKG_CHAINLOAD_TOOLCHAIN_FILE": "C:/d/WindowsToolchain/Windows.MSVC.toolchain.cmake",
		"VCPKG_DUMPBIN": "C:/Program Files/Microsoft Visual Studio/18/Community/VC/Tools/MSVC/14.50.35717/bin/Hostx64/x64/dumpbin.exe"
      }
    }
  ]
}
```

For configure and build I run:

```powershell
cmake --preset=vcpkg-msvc -B build 
cmake --build build
```

All needed dlls are copied to the build folder as part of the build process.
